### ISSUE TITLE
preset: 'client' is creating the same type twice

### ISSUE THREAD

**Issue by jcollum-nutrien** [dotansimha/graphql-code-generator#8571]
### Describe the bug

codegen with preset: 'client'  is creating the same types twice

### Your Example Website or App

see additional context below

### Steps to Reproduce the Bug or Issue

1. save config file as config.ts 
1. run `graphql-codegen --config config.ts` 
2. run `cat src/generated/faker-olympus/graphql.ts | grep -n 'export type Card '

### Expected behavior

`type Card` is created once 

Actual results: 

```
$ cat src/generated/faker-olympus/graphql.ts | grep -n 'export type Card '
20:export type Card = Nameable & {
166:export type Card = Nameable & {`
```

### Screenshots or Videos

![image](https://user-images.githubusercontent.com/99687439/199851695-a9c8faab-1ccb-408a-82ec-8e40a6eb113f.png)

![image](https://user-images.githubusercontent.com/99687439/199851705-ed091265-80ee-42c5-8d24-fb48f11a11f2.png)

### Platform

Mac, Node 14. See other versions below. 

```
"devDependencies": {
    "@graphql-codegen/cli": "~2.13.8",
    "@graphql-codegen/client-preset": "~1.1.1",
    "@graphql-codegen/introspection": "~2.2.1",
    "@graphql-codegen/typescript": "~2.8.0",
    "@graphql-codegen/typescript-resolvers": "~2.7.5",
    "prettier": "2.7.1",
    "typescript": "4.8.4"
  },
```

### Codegen Config File

import type {CodegenConfig} from '@graphql-codegen/cli';

const config: CodegenConfig = {
  overwrite: true,
  schema: "https://faker.graphqleditor.com/a-team/olympus/graphql",
  generates: {
    "src/generated/faker-olympus/": {
      preset: 'client',
      plugins: ["typescript", "typescript-resolvers"]
    }
  }
};

export default config;

### Additional context

I'll be putting this in a repo soon...

```
{
  "name": "gql-gen-olympus",
  "description": "sample repo for generating and using generated gql code",
  "license": "MIT",
  "scripts": {
    "codegen-faker": "graphql-codegen --config config/faker-olympus-codegen.ts"
  },
  "dependencies": {
    "@apollo/client": "3.7.1",
    "graphql": "16.6.0",
    "graphql-request": "5.0.0"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "~2.13.8",
    "@graphql-codegen/client-preset": "~1.1.1",
    "@graphql-codegen/introspection": "~2.2.1",
    "@graphql-codegen/typescript": "~2.8.0",
    "@graphql-codegen/typescript-resolvers": "~2.7.5",
    "prettier": "2.7.1",
    "typescript": "4.8.4"
  },
  "prettier": {
    "trailingComma": "es5",
    "singleQuote": true,
    "printWidth": 80,
    "semi": true
  }
}

```

**Comment by jcollum-nutrien** [comment]
Repo: https://github.com/jcollum-nutrien/gql-code-generation-olympus

**Comment by jkepps** [comment]
I could be wrong, but I don't think you need the typescript plugins when you're using the `client` preset

**Comment by jcollum-nutrien** [comment]
> I could be wrong, but I don't think you need the typescript plugins when you're using the `client` preset

`plugins` property is required though.

**Comment by jkepps** [comment]
yeah it's a little misleading to say that the plugins property is required even though it can be set to an empty array. would probably make more sense to make it optional. glad that worked for you though!

**Comment by loicnestler** [comment]
Having the same issue with the following config (event though I have `plugins` set):

<img width="457" alt="image" src="https://user-images.githubusercontent.com/48695625/214477344-187f5bca-eea4-4018-91ab-f8e8e41cb4cf.png">

**Comment by exsesx** [comment]
+1 with plugins, empty array "solves" the issue.

**Comment by yudiz-bhoomit** [comment]
But what if I want to generate queries and mutations it won't get generated right, if I remove the plugins?
How can I solve that if I want to use plugins and want to continue ?

**Comment by codingwesley** [comment]
+1 same issue

**Comment by eddeee888** [comment]
Hi all 👋 
Client Preset should handle all types without the need to add extra plugins (The plugins array is now optional).
We now have this guide: https://the-guild.dev/graphql/codegen/plugins/presets/preset-client
Please let me know if it still doesn't work after following the guide 🙂
